### DISCUSSION TITLE
How can I change the icon colorscheme?

### DISCUSSION THREAD

**Discussion by sainttttt** [ibhagwan/fzf-lua#1455] [Q&A] [answered]
Hi! I was wondering if there is a way to change the colorscheme of the icons (dev icons or mini icons) that are used it fzf lua. https://github.com/dgox16/devicon-colorscheme.nvim - this plugin seems to change the colorscheme in some parts of nvim but it doesn't seem to carry over for fzf-lua. Any help would be appreciated. Thanks!

**Comment by ibhagwan** [comment]
Haven’t looked into it’s internal but if it works the way I think it does calling `require("fzf-lua.devicons").unload()` once (after changing the theme) might work.

**Reply by sainttttt** [reply]
Tried doing this but didn't seem to work.
```
  {
    "dgox16/devicon-colorscheme.nvim",
    dependencies = {
      "nvim-tree/nvim-web-devicons",
    },
    config = function()
      require("devicon-colorscheme").setup({
        colors = {
          blue = "#92a2d5",
          cyan = "#85b5ba",
          green = "#000000",
          magenta = "#e29eca",
          orange = "#f5a191",
          purple = "#aca1cf",
          red = "#ea83a5",
          white = "#c9c7cd",
          yellow = "#ea83a5",
          bright_blue = "#000000",
          bright_cyan = "#99c9ce",
          bright_green = "#9dc6ac",
          bright_magenta = "#ecaad6",
          bright_orange = "#ffae9f",
          bright_purple = "#b9aeda",
          bright_red = "#f591b2",
          bright_yellow = "#ea83a5",
        },
      })
      require("fzf-lua.devicons").unload()
    end
  },
```

Tried this but the changes don't take effect in fzf-lua. The icons do change color in other parts of nvim though!

**Reply by ibhagwan** [reply]
Fzf-lua gets the colors directly from the internal structures of nvim-web-devicons using the API, I have t explored the code of the plug-in you listed but it’s possible it’s only changing the DevIconExt highlight groups.

**Reply by sainttttt** [reply]
Yeah https://github.com/dgox16/devicon-colorscheme.nvim/blob/fdfce12e3c40b0bcde82d09e75ac1ac72d8095fe/lua/devicon-colorscheme/highlights.lua#L12 That is what it seems like its doing.

**Reply by ibhagwan** [reply, marked as answer]
The devicons code has been through rounds of optimizations in order to get correct colors in the external process (icons customizations, light/dark modes), unless there’s official support through the original plugins which changes the colors returned by the API I don’t think it’s worth the effort.

Try mini.icons it should have an adaptable color scheme which also gets updated from the highlights on every run.

**Comment by sainttttt** [comment]
Okay I'll look into that then, thanks!
